pkg/resources/table.go

```diff
@@ -786,14 +807,18 @@ func UpdateTable(d *schema.ResourceData, meta interface{}) error {
 				addRequest.WithComment(sdk.String(cA.comment))
 			}
 
+			if cA.collate != "" && strings.Contains(cA.dataType, "CHAR") || cA.dataType == "STRING" || cA.dataType == "TEXT" {
+				addRequest.WithCollate(sdk.String(cA.collate))
+			}
+
 			err := client.Tables.Alter(ctx, sdk.NewAlterTableRequest(id).WithColumnAction(sdk.NewTableColumnActionRequest().WithAdd(addRequest)))
 			if err != nil {
 				return fmt.Errorf("error adding column: %w", err)
 			}
 		}
 		for _, cA := range changed {
-			if cA.changedDataType {
-				err := client.Tables.Alter(ctx, sdk.NewAlterTableRequest(id).WithColumnAction(sdk.NewTableColumnActionRequest().WithAlter([]sdk.TableColumnAlterActionRequest{*sdk.NewTableColumnAlterActionRequest(fmt.Sprintf("\"%s\"", cA.newColumn.name)).WithType(sdk.Pointer(sdk.DataType(cA.newColumn.dataType)))})))
+			if cA.changedDataType || cA.changedCollate {
+				err := client.Tables.Alter(ctx, sdk.NewAlterTableRequest(id).WithColumnAction(sdk.NewTableColumnActionRequest().WithAlter([]sdk.TableColumnAlterActionRequest{*sdk.NewTableColumnAlterActionRequest(fmt.Sprintf("\"%s\"", cA.newColumn.name)).WithType(sdk.Pointer(sdk.DataType(cA.newColumn.dataType))).WithCollate(sdk.String(cA.newColumn.collate))})))
```

Comment:
Afaik column collation cannot be changed (also couldn't find anything in the documentation). There's no `alter column x set collation` I tried to set data type with the same data type and different collation, but Snowflake returned `SQL compilation error: cannot change column S from type "VARCHAR(16777216) COLLATE 'utf8'" to "VARCHAR(16777216) COLLATE 'en'" because they have incompatible collations.`. Maybe I'm not aware of something, but It should also show when you add test case I proposed.

Reply:
Thank you for your feedback! I agree with you.
However, if the collate change is on terraform, I think we need to tell the user in some way that the collate change will fail.
I would think that changing a VARCHAR to a shorter length, for example, would fail as well, but I didn't think it would be a problem since it would behave the same way as this implementation. What do you think?

Reply:
Yeah, you're right. Let's leave it as is.